main.go

```diff
@@ -92,13 +94,17 @@ func (d *discovery) Run(ctx context.Context, ch chan<- []*targetgroup.Group) {
 	}
 }
 
-func newDiscovery(conf sdConfig) (*discovery, error) {
-	cd := &discovery{
```

Comment:
Based on other comments, it can be more simple.

```
func newDiscovery(conf sdConfig) (*discovery, error) {
	cd := &discovery{
		apiUrl:          conf.ApiUrl,
		outputFile:      conf.OutputFile,
		refreshInterval: conf.RefreshInterval,
		logger:          logger,
	}
	return cd, nil
}
```